### ISSUE TITLE
ChatGLM能做增量训练吗

### ISSUE THREAD

**Issue by Bob-Annette** [THUDM/ChatGLM-6B#1239]
### Solutions

以ChatGLM为基座，增加自己的语料库进行预训练，不是微调，这个该怎么做

**Comment by guotong1988** [comment]
同问

**Comment by JiaXiao243** [comment]
同问

**Comment by ljch2018** [comment]
+1

**Comment by Liuxc2005** [comment]
同问

**Comment by JiaXiao243** [comment]
同问

**Comment by runzhi214** [comment]
应该是不能

**Comment by NewNLPer** [comment]
I am currently working on this issue. I have previously dealt with incremental pre-training for llama, and the key to the problem is the function of  data_collator processing.

**Comment by ArtificialZeng** [comment]
> I am currently working on this issue. I have previously dealt with incremental pre-training for llama, and the key to the problem is the function of data_collator processing.

现在解决了吗

**Comment by NewNLPer** [comment]
Yes, I have completed it, but some unexpected situations have occurred and the model has experienced catastrophic forgetting. If it is not particularly necessary, I do not recommend doing so.

**Comment by starphantom666** [comment]
> 是的，我已经完成了，但是发生了一些类似的情况，模型经历了灾难性的遗忘。如果不是特别有必要的事情，我不建议这样做。

但是低资源想要训练领域模型好像只能这样做

**Comment by Alfer-Feng** [comment]
> > 是的，我已经完成了，但是发生了一些类似的情况，模型经历了灾难性的遗忘。如果不是特别有必要的事情，我不建议这样做。
> 
> 但是低资源想要训练领域模型好像只能这样做

>低资源是指只有无监督预料，无有监督数据吗？

**Comment by guoqiangqi** [comment]
@NewNLPer 你好，我们也在计划这样做，方便请教下如何修改脚本吗？

**Comment by NewNLPer** [comment]
Yes, I need to reiterate that what I am doing is incremental training. Unlike fine-tuning, the purpose of fine-tuning is to make LLM understand instructions, but the purpose of incremental training is to increase LLM's domain knowledge. The general approach to pulling down the vertical domain of LLM is (in conjunction with Lora):
**1. LLM+fine tuning**
**2. LLM+incremental training+fine tuning**

I can provide an example to understand the training method of LLM:
**Fine tuning:**
**Input_ Id**: [1,2,3,4]
**labels**: [5,6,7,8]
**Cat_ Tensor**: [-100, -100, -100, -100,5,6,7,8]
**Incremental training:**
**Input**: [1,2,3,4]
**labels**: [5,6,7,8]
**Cat_ Tensor**: [1,2,3,4,5,6,7,8]
For example, your best choice is to take a look at the data_ Collector,observe the input_ ids and labels ,You can understand it.

**Comment by guoqiangqi** [comment]
Thank you for your reply.
I understand the difference between incremental pretraining and fine-tuning
that you mentioned. But I'm not very familiar with the data processing and
model parameter update pipeline of LLMs. Is it convenient for you to
share me a training script (perhaps a trainer)? I would greatly appreciate
it !

NewNLPer ***@***.***> 于2023年7月14日周五 16:55写道：
